src/strategies/dss-vest-unpaid/index.ts

```diff
@@ -0,0 +1,55 @@
+import { BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Contract } from '@ethersproject/contracts';
+import { Multicaller } from '../../utils';
+
+export const author = 'espendk';
+export const version = '1.0.0';
+
+const abi = [
+  'function ids() external view returns (uint256)',
+  'function usr(uint256 id) external view returns (address)',
+  'function unpaid(uint256 id) external view returns (uint256)'
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  // Get the number of vestings
+  const dssVestContract = new Contract(options.address, abi, provider);
+  const idCount = await dssVestContract.ids();
+
+  // Get the vesting addresses and unpaid amounts
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+  for (let id = 1; id <= idCount; ++id) {
```

Comment:
Could we add a limit to idCount here, to avoid memory issues in future, you can return an error if we get more than what you expect

Reply:
Sure. Added in [`369cbdc` (#1290)](https://github.com/snapshot-labs/snapshot-strategies/pull/1290/commits/369cbdc8442cbecac1151cf6bd19fbbe5c771b1f)